onnxruntime/contrib_ops/cpu/qlinear_lookup_table.cc

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "qlinear_lookup_table.h"
+#include <limits>
+#include <cmath>
+
+namespace onnxruntime {
+namespace contrib {
+
+static void QLinearLookupTableTransform(const uint8_t* x, const uint8_t table[256], uint8_t* y, size_t n) {
+  for (; n >= 4; n -= 4) {
+    const size_t x_value0 = x[0];
+    const size_t x_value1 = x[1];
+    const size_t x_value2 = x[2];
+    const size_t x_value3 = x[3];
+    x += 4;
+    const uint8_t table_value0 = table[x_value0];
+    const uint8_t table_value1 = table[x_value1];
+    const uint8_t table_value2 = table[x_value2];
+    const uint8_t table_value3 = table[x_value3];
+
+    y[0] = table_value0;
+    y[1] = table_value1;
+    y[2] = table_value2;
+    y[3] = table_value3;
+    y += 4;
+  }
+  for (; n != 0; --n) {
+    const size_t x_value0 = *x++;
+    const uint8_t table_value0 = table[x_value0];
+    *y++ = table_value0;
+  }
+}
+
+template <typename T>
+static void BuildQLinearLookupTable(
+    uint8_t table[256], const float X_scale, const T X_zero_point, const float Y_scale, const T Y_zero_point, float alpha) {
+  constexpr int qmin = std::numeric_limits<T>::min();
+  constexpr int qmax = std::numeric_limits<T>::max();
+  for (int i = 0; i < 256; ++i) {
+    T x = static_cast<T>(i);
+    float x_dequantized = X_scale * (static_cast<int>(x) - static_cast<int>(X_zero_point));
+    float y = x_dequantized >= 0.0f ? x_dequantized : alpha * x_dequantized;
+    int y_quantized = static_cast<int>(std::round(y / Y_scale)) + static_cast<int>(Y_zero_point);
```

Comment:
The math here is different from using MlasQuantizeLinear: the use of std::round instead of std::nearbyintf + plus what happens if the float number is outside of the quantization range. Instead, could you just have a stack buffer of say 16 floats and do it batches and call MlasQuantizeLinear to let it use the same logic used elsewhere?

Reply:
Good Idea to use the MlasQuantizeLinear() for consistency. Will change it.